R/model.R

```diff
@@ -844,3 +844,61 @@ variational_method <- function(data = NULL,
   CmdStanVB$new(runset)
 }
 CmdStanModel$set("public", name = "variational", value = variational_method)
+
+#' Run Stan's standalone generated quantities method
+#'
+#' @name model-method-generate-quantities
+#' @aliases generate_quantities
+#' @family CmdStanModel methods
+#'
+#' @description The `$generate_quantities()` method of a [`CmdStanModel`] object runs
+#'   Stan's standalone generated quantites to obtain generated quantities based on
+#'   previously fitted parameters.
+#'
+#' @details 
+#'
+#'   -- [*CmdStan Interface User's Guide*](https://github.com/stan-dev/cmdstan/releases/latest)
+#'
+#' @section Usage:
+#'   ```
+#'   $generate_quantities(
+#'     fitted_params = NULL,
+#'     data = NULL
+#'   )
+#'   ```
+#'
+#' @template model-common-args
+#' @section 
+#'
+#' @section Value: The `$generate_quantities()` method returns a [`CmdStanGQ`] object.
+#'
+#' @template seealso-docs
+#' @inherit cmdstan_model examples
+#'
+NULL
+
+generate_quantities_method <- function(fitted_params = NULL,
+                            data = NULL,
+                            seed = NULL,
+                            refresh = NULL,
+                            output_dir = NULL) {
+  generate_quantities_args <- GenerateQuantitiesArgs$new(
+    fitted_params = fitted_params # TODO(process_fitted_params)
```

Comment:
cmdstanpy allows fitted_params to be a path to the csv or the fit object. I think we should also allow the result of fit$draws() to be allowed here (but maybe in a separate PR).

In case a fit object is used here, it makes sense to check if the CSV files still exist in which case just use those and if not write the draws to a temp CSV file. Does that sound right?

Reply:
Yeah I think that sounds right.